dotCMS/src/main/java/com/dotcms/keyvalue/business/KeyValueAPIImpl.java

```diff
@@ -0,0 +1,182 @@
+        if (null != results && !results.isEmpty()) {
+            this.cache.add(key, results);
+        }
+        return results;
+    }
+    @Override
+    public List<KeyValue> get(final String key, final User user, final boolean respectFrontendRoles) {
+        List<KeyValue> results = this.cache.get(key);
+        if (null != results) {
+            return results;
+        }
+        results = queryKeyValues(key, -1, null, user, respectFrontendRoles);
+        if (null != results && !results.isEmpty()) {
+            this.cache.add(key, results);
+        }
+        return results;
+    }
+
+
+    @Override
+    public List<KeyValue> get(final String key, final ContentType contentType,
+                              final long languageId, final User user, final boolean respectFrontendRoles) {
+        List<KeyValue> results = this.cache.getByLanguage(key, languageId);
+        if (null != results) {
+            return results;
+        }
+        results = queryKeyValues(key, languageId, contentType, user, respectFrontendRoles);
+        if (null != results && !results.isEmpty()) {
+            this.cache.addByLanguage(key, languageId, results);
+        }
+        return results;
+    }
+
+    @Override
+    public List<KeyValue> get(final String key, final long languageId, final User user, final boolean respectFrontendRoles) {
+        List<KeyValue> results = this.cache.getByLanguage(key, languageId);
+        if (null != results) {
+            return results;
+        }
+        results = queryKeyValues(key, languageId, null, user, respectFrontendRoles);
+        if (null != results && !results.isEmpty()) {
+            this.cache.addByLanguage(key, languageId, results);
+        }
+        return results;
+    }
+
+    @Override
+    public List<KeyValue> get(final String key, final ContentType contentType, final User user,
+                    final boolean respectFrontendRoles) {
+        List<KeyValue> results = this.cache.getByContentType(key, contentType.id());
+        if (null != results) {
+            return results;
+        }
+        results = queryKeyValues(key, -1, contentType, user, respectFrontendRoles);
+        if (null != results && !results.isEmpty()) {
+            this.cache.addByContentType(key, contentType.id(), results);
+        }
+        return results;
+    }
+
+    @Override
+    public KeyValue get(final String key, final long languageId, final ContentType contentType, final User user,
+                    final boolean respectFrontendRoles) {
+        if (languageId > -1 && null != contentType && UtilMethods.isSet(contentType.id())) {
+            final KeyValue keyValue = this.cache.getByLanguageAndContentType(key, languageId, contentType.id());
+            if (null != keyValue) {
+                return keyValue;
+            }
+        }
+
+        final List<KeyValue> result = queryKeyValues(key, languageId, contentType, user, respectFrontendRoles);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "queryKeyValues()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)